modules/search/class.jetpack-search.php

```diff
@@ -102,24 +102,49 @@ public function init_hooks() {
 	 * @return object|WP_Error The response from the public api, or a WP_Error
 	 */
 	public function search( array $es_args ) {
-		$service_url = 'https://public-api.wordpress.com/rest/v1/sites/' . $this->jetpack_blog_id . '/search';
+		$endpoint    = sprintf( '/sites/%s/search', $this->jetpack_blog_id );
+		$service_url = 'https://public-api.wordpress.com/rest/v1' . $endpoint;
 
-		$start_time = microtime( true );
+		$do_authenticated_request = false;
+
+		if ( class_exists( 'Jetpack_Client' ) &&
+			isset( $es_args['authenticated_request'] ) &&
```

Comment:
If we know that `$es_args['authenticated_request']` will assume boolean values, we could shorten this to:

`$do_authenticated_request = class_exists( 'Jetpack_Client' ) && ! empty( $es_args['authenticated_request'] );`

Reply:
@vindl Thanks for looking! 

IMO it's better to be explicit and use strict equality, due to type juggling in PHP...since the string `"false"` evaluates to `true` and is not empty for example.

This way makes it a little harder to `_do_it_wrong()` as _only_ the expected value of `true` will cause the behavior. Since doing this could potentially expose otherwise non-public posts, it should be (relatively) hard to trigger. 

It's better if `$es_args['authenticated_requests'] = 'no';`, `$es_args['authenticated_requests'] = 'bananas';`, etc don't work :)

Reply:
I also have to give credit to @mjangda who wrote the original code this was ported from.

Reply:
It did occur to me while reviewing but I thought we can guarantee that that value will always be boolean. I agree with you though, better to be safe here than expose ourselves to that risk with the sole aim of saving a couple of lines.